glustercli/cmd/snapshot-delete.go

```diff
@@ -19,24 +21,78 @@ var (
 		Use:   "delete <snapname>",
 		Short: snapshotDeleteHelpShort,
 		Long:  snapshotDeleteHelpLong,
-		Args:  cobra.ExactArgs(1),
+		Args:  cobra.RangeArgs(1, 3),
 		Run:   snapshotDeleteCmdRun,
 	}
 )
 
+var (
+	snapshotDeleteAllCmd = &cobra.Command{
+		Use:   "all [volname]",
+		Short: snapshotDeleteAllHelpShort,
+		Long:  snapshotDeleteAllHelpLong,
+		Args:  cobra.MaximumNArgs(1),
+		Run:   snapshotDeleteAllCmdRun,
+	}
+)
+
 func init() {
 	snapshotCmd.AddCommand(snapshotDeleteCmd)
+	snapshotDeleteCmd.AddCommand(snapshotDeleteAllCmd)
 }
 
-func snapshotDeleteCmdRun(cmd *cobra.Command, args []string) {
-	snapname := args[0]
-
+func snapshotDelete(snapname string) error {
 	if err := client.SnapshotDelete(snapname); err != nil {
 		if GlobalFlag.Verbose {
 			log.WithError(err).WithField(
 				"snapshot", snapname).Error("snapshot delete failed")
 		}
-		failure("Snapshot delete failed", err, 1)
+		return err
 	}
 	fmt.Printf("%s Snapshot deleted successfully\n", snapname)
+	return nil
+}
+
+func snapshotDeleteCmdRun(cmd *cobra.Command, args []string) {
+
+	if len(args) != 1 {
+		//snapshotDeleteAllCmdRun will take care of else case.
```

Comment:
@rafikc30 just a question, `snapshotDeleteAllCmdRun` will take care if `len(args) != 1` right?
if yes, should this comment need to be updated?

Reply:
yes, you are right